dex-it/src/test/scala/com/wavesplatform/it/sync/smartcontracts/AllowedBlockchainStateAccountsTestSuite.scala

```diff
@@ -0,0 +1,123 @@
+
+import scala.concurrent.duration._
+
+final class AllowedBlockchainStateAccountsTestSuite extends MatcherSuiteBase {
+
+  "AllowedBlockchainStateAccountsTestSuite" - {
+
+    "should fail while accessing blockchain state from usual accounts" in {
+      updateAccountScript(bob, script1)
+      val order = mkOrder(bob, wavesUsdPair, OrderType.SELL, 10.waves, 1, version = 2)
+      dex1.tryApi.place(order) should failWith(
+        AccountScriptReturnedError.code,
+        "The account's script of 3Q6ujVDbX57oLsXxifqfTcycgb4S8U3DLFz returned the error: An access to the blockchain.height is denied on DEX"
+      )
+    }
+
+    "should succeed while accessing blockchain state from custom accounts" in {
+      updateAccountScript(carol, script1)
+      val carolOrder = mkOrder(carol, wavesUsdPair, OrderType.SELL, 10.waves, 1, version = 2)
+      dex1.api.place(carolOrder)
+      val aliceOrder = mkOrder(alice, wavesUsdPair, OrderType.BUY, 10.waves, 1, version = 2)
+      placeAndAwaitAtDex(aliceOrder, HttpOrderStatus.Status.Filled)
+      val txId = ByteStr(dex1.api.getTransactionsByOrderId(carolOrder).head.id().bytes())
+      eventually {
+        wavesNode1.tryApi.transactionInfo(txId).isRight shouldBe true
+      }
+    }
+
+    "transaction should be rejected by a new script which will be set after placing an order" in {
+      updateAccountScript(carol, script1)
+      val carolOrder = mkOrder(carol, wavesUsdPair, OrderType.SELL, 10.waves, 1, version = 2)
+      dex1.api.place(carolOrder)
+      updateAccountScript(carol, script2)
+      val aliceOrder = mkOrder(alice, wavesUsdPair, OrderType.BUY, 10.waves, 1, version = 2)
+      placeAndAwaitAtDex(aliceOrder, HttpOrderStatus.Status.Filled)
+      val txId = ByteStr(dex1.api.getTransactionsByOrderId(carolOrder).head.id().bytes())
+      Thread.sleep(5.seconds.toMillis)
+      wavesNode1.tryApi.transactionInfo(txId).isLeft shouldBe true
+    }
+
+    "order should be rejected by script" in {
+      updateAccountScript(carol, script2)
+      val carolOrder = mkOrder(carol, wavesUsdPair, OrderType.SELL, 10.waves, 1, version = 2)
+      val placeResult = dex1.tryApi.place(carolOrder)
+      placeResult should failWith(
+        AccountScriptDeniedOrder.code,
+        "The account's script of 3Q97CnwDv7pE9wkYhEAq3juftoJiH1eaHGk rejected the order"
+      )
+    }
+  }
+
+  override protected def beforeAll(): Unit = {
+    wavesNode1.start()
+    broadcastAndAwait(IssueUsdTx)
+    broadcastAndAwait(mkTransfer(alice, carol, 1000.waves, Waves))
+    dex1.start()
+  }
+
+  override protected val dexInitialSuiteConfig: Config = ConfigFactory
+    .parseString(
+      s"""waves.dex {
+         |  price-assets = [ "$UsdId", "WAVES" ]
+         |}""".stripMargin
+    )
+
+  override protected def wavesNodeInitialSuiteConfig: Config = ConfigFactory
+    .parseString(
+      s"""
+         |waves.dex.order-script-validation.allowed-blockchain-state-accounts=["${carol.publicKey.base58}"]
```

Comment:
How about to create an alias for `carol` at the top of the class and use it in the test suite? E.g.:
```scala
val allowedAcc = carol
```

Reply:
fxd